### ISSUE TITLE
save_files test coverage

### ISSUE THREAD

**Issue by DJuboor** [lbanawa/photo_filters#5]
**TL;DR:**
Write tests for the save_files function and update the function to handle these test cases when applicable.

**Objective:**
In src/processing.py, identify any _reasonable_ unhandled edge cases that may break the save_files function. Modify the save_files function to better handle these failures by returning a False status from the function and a helpful message to the user. Write unittests in tests/test_process.py to simulate an input that would cause the edge case. Finally, write a test case that runs the function as expected.

**Suggested Tests:**
Fail Cases:
- [x] filtered_photos_to_save is a handles invalid inputs (not a list / an empty list / is a list of non-arrays)
- [x] filtered_photos_to_save and file_names are not the same length
- [ ] file_names is not a list / is an empty list 
- [ ] output_folder is not a string
- [ ] output_folder is not a valid folder path
- [ ] output_folder is a valid folder path, but the folder does not exist

Success Cases:
- [ ] all inputs are as expected, files are saved as expected with the expected file name.

**Success criteria:**
- all ^ test cases are well handled. 
- all ^ tests written in tests/test_process.py and able to be successfully run via run_tests.py

**Comment by lbanawa** [comment]
67a13895b8cc95db27e23688546cf80312dc95b4

Test for filtered_photos_to_save as invalid inputs

**Comment by lbanawa** [comment]
8745a198c5984ba54604e07870a681797997690c

Test for filtered_photos_to_save and file_names as different list lengths

**Comment by lbanawa** [comment]
f297073b55e85bc14ebc85c5b059c6bbc72f0050

Final test suite
Updated save_files and run_filter functions and tests
Clean up

**Comment by lbanawa** [comment]
Mereged pull request: https://github.com/lbanawa/photo_filters/pull/9
Additional tests are out of scope
